### PR TITLE
Add searching by location and description

### PR DESCRIPTION
**Pull request by Rekfuki**

### CODE REVIEW

File: api/db/helper.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	if err != nil {
+		return 0, err
+	}
+
+	return total, nil
+}
+
+func (c *Client) paginate(query string, pageNumber int, perPage int) string {
+	if pageNumber > 0 && perPage > 0 {
+		// TODO Change to use placeholders, go-xorm/builder should support it soon
```
Comment (Watsakasaurus) [inline api/db/helper.go]: needs to be addressed before final product launch

File: api/db/helper.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+type ILike [2]string
+
+var _ builder.Cond = ILike{"", ""}
+
+// WriteTo write SQL to Writer
+func (ilike ILike) WriteTo(w builder.Writer) error {
+	if _, err := fmt.Fprintf(w, "%s ILIKE ?", ilike[0]); err != nil {
+		return err
+	}
+	// FIXME: if use other regular express, this will be failed. but for compatible, keep this
```
Comment (Watsakasaurus) [inline api/db/helper.go]: needs to be resolved before product launch

File: api/db/helper.go (hunk 1 above)
Comment (Rekfuki) [reply api/db/helper.go]: It will be changed once it's oficially supported.

File: api/db/helper.go (hunk 2 above)
Comment (Rekfuki) [reply api/db/helper.go]: It's unlikely we will need to support regex, but it's left just for reference.

File: api/db/helper.go (hunk 1 above)
Comment (Watsakasaurus) [reply api/db/helper.go]: all okay, long term to do, will keep up to date to implement when supported

File: api/db/helper.go (hunk 2 above)
Comment (Watsakasaurus) [reply api/db/helper.go]: extensive security procedure, not a priority

**Comment by Watsakasaurus** [comment]
issues resolved, ready to merge

**Pull request merged**